be/src/exec/iceberg/iceberg_delete_builder.cpp

```diff
@@ -37,63 +38,147 @@ static const IcebergColumnMeta k_delete_file_path{
-    auto iter = std::make_unique<IcebergDeleteFileIterator>();
-    RETURN_IF_ERROR(iter->init(_fs, timezone, delete_file_path, file_length, slot_descriptors, true));
-    std::shared_ptr<::arrow::RecordBatch> batch;
+StatusOr<std::unique_ptr<RandomAccessFile>> IcebergDeleteBuilder::open_random_access_file(
+        const TIcebergDeleteFile& delete_file, HdfsScanStats& fs_scan_stats, HdfsScanStats& app_scan_stats,
+        std::shared_ptr<io::SharedBufferedInputStream>& shared_buffered_input_stream,
+        std::shared_ptr<io::CacheInputStream>& cache_input_stream) const {
+    const OpenFileOptions options{.fs = _params.fs,
+                                  .path = delete_file.full_path,
+                                  .file_size = delete_file.length,
+                                  .fs_stats = &fs_scan_stats,
+                                  .app_stats = &app_scan_stats,
+                                  .datacache_options = _params.datacache_options};
+    ASSIGN_OR_RETURN(auto file,
+                     HdfsScanner::create_random_access_file(shared_buffered_input_stream, cache_input_stream, options));
+    std::vector<io::SharedBufferedInputStream::IORange> io_ranges{};
+    for (int64_t offset = 0; offset < delete_file.length;) {
+        const int64_t remain_length =
+                std::min(static_cast<int64_t>(config::io_coalesce_read_max_buffer_size), delete_file.length - offset);
+        io_ranges.emplace_back(offset, remain_length);
+        offset += remain_length;
+    }
 
-    Status status;
-    while (true) {
-        status = iter->has_next();
-        if (!status.ok()) {
-            break;
-        }
+    RETURN_IF_ERROR(shared_buffered_input_stream->set_io_ranges(io_ranges));
+    return file;
+}
 
-        batch = iter->next();
-        ::arrow::StringArray* file_path_array = static_cast<arrow::StringArray*>(batch->column(0).get());
-        ::arrow::Int64Array* pos_array = static_cast<arrow::Int64Array*>(batch->column(1).get());
-        for (size_t row = 0; row < batch->num_rows(); row++) {
-            if (file_path_array->Value(row) == _datafile_path) {
-                need_skip_rowids->emplace(pos_array->Value(row));
-            }
+Status IcebergDeleteBuilder::fill_skip_rowids(const ChunkPtr& chunk) const {
+    const ColumnPtr& file_path = chunk->get_column_by_slot_id(k_delete_file_path.id);
+    const ColumnPtr& pos = chunk->get_column_by_slot_id(k_delete_file_pos.id);
+    for (int i = 0; i < chunk->num_rows(); i++) {
+        if (file_path->get(i).get_slice() == _params.path) {
+            _need_skip_rowids->emplace(pos->get(i).get_int64());
         }
     }
+    return Status::OK();
+}
+
+Status IcebergDeleteBuilder::build_parquet(const TIcebergDeleteFile& delete_file) const {
+    HdfsScanStats app_scan_stats;
+    HdfsScanStats fs_scan_stats;
+    std::shared_ptr<io::SharedBufferedInputStream> shared_buffered_input_stream;
+    std::shared_ptr<io::CacheInputStream> cache_input_stream;
```

Comment:
how about disabled cache?

Reply:
it's up to config. default is null and I have checked it.